src/lib.rs

```diff
@@ -212,56 +242,62 @@ impl<O> IO<O> {
 /// assert_eq!(result, "meow\n");
 /// # Ok (()) }
 /// ```
-pub fn pure <T> (value: T) -> IO<T> {
+pub fn pure <'a, T> (value: T) -> IO<'a, T> where T: 'a {
     IO (Pure (value))
 }
 
 /// Ignore the output value of `x`.
-pub fn void <T> (x: IO<T>) -> IO<()>
+pub fn void <'a, T> (x: IO<'a, T>) -> IO<'a>
 where
     T: 'static
 {
     x.replace(())
 }
 
 /// Print a line to stdout.
-pub fn println (s: impl ToString) -> IO<()> {
+pub fn println (s: impl ToString) -> IO<'static> {
     struct PrintLn (String);
 
-    impl Action for PrintLn {
+    impl Action<'static> for PrintLn {
         type Output = ();
         fn perform (&mut self, Context { o, .. }: Context<'_>) -> Result<Self::Output> {
             writeln!(o, "{}", self.0)
         }
     }
-    
+
     IO::new(PrintLn (s.to_string()))
```

Comment:
Could you avoid having the IO take ownership of `s` by making it's lifetime not static?